src/level/robotmain.cpp

```diff
@@ -5482,93 +5483,47 @@ void CRobotMain::SetAutosaveSlots(int slots)
     if (m_autosaveSlots == slots) return;
 
     m_autosaveSlots = slots;
-    AutosaveRotate(false);
+    AutosaveRotate();
 }
 
 int CRobotMain::GetAutosaveSlots()
 {
     return m_autosaveSlots;
 }
 
-int CRobotMain::AutosaveRotate(bool freeOne)
+// Remove oldest saves with autosave prefix
+void CRobotMain::AutosaveRotate()
```

Comment:
I guess `AutosaveRotate()` call can be removed when setting slot number. This way it'll remain simple and it'll delete old saves on next autosave.